packages/app/components/stackbox/FrequencyOptionsCard.tsx

```diff
@@ -0,0 +1,132 @@
+"use client";
+
+import { ChangeEvent, useEffect, useState } from "react";
+
+import { twMerge } from "tailwind-merge";
+
+import { FREQUENCY_OPTIONS } from "@/models";
+import { BodyText, RadioButton, TextInput } from "@/ui";
+
+interface FrequencyOptionsCardProps {
+  frequency: FREQUENCY_OPTIONS;
+  setEndDate: (date: Date) => void;
+}
+
+const defaultFrequencyOptions = {
+  [FREQUENCY_OPTIONS.hour]: ["6", "12", "24"],
+  [FREQUENCY_OPTIONS.day]: ["7", "15", "30"],
+  [FREQUENCY_OPTIONS.week]: ["4", "8", "12"],
+  [FREQUENCY_OPTIONS.month]: ["2", "6", "12"],
+};
+const maxCustomFrequencies = {
+  [FREQUENCY_OPTIONS.hour]: 96,
+  [FREQUENCY_OPTIONS.day]: 365,
+  [FREQUENCY_OPTIONS.week]: 52,
+  [FREQUENCY_OPTIONS.month]: 24,
+};
+const postiveIntegerOnly = /^$|^[0-9]+$/;
+
+const getEndDateByDefaultFrequency = (
+  frequency: FREQUENCY_OPTIONS,
+  timeAmount: number
+) => {
+  const dateNow = new Date();
+  const endDateByDefaultFrequency = {
+    [FREQUENCY_OPTIONS.hour]: dateNow.setHours(dateNow.getHours() + timeAmount),
+    [FREQUENCY_OPTIONS.day]: dateNow.setDate(dateNow.getDate() + timeAmount),
+    [FREQUENCY_OPTIONS.week]: dateNow.setDate(
+      dateNow.getDate() + timeAmount * 7
+    ),
+    [FREQUENCY_OPTIONS.month]: dateNow.setMonth(
+      dateNow.getMonth() + timeAmount
+    ),
```

Comment:
We need to have this object outside the function, and returning a function itself. Otherwise we all calculating all this different values for hour, days,..., month when we are only using 1 value. Also it creates a bug, where we are adding `timeAmount` to `dateNow`, so when using months for example, we are adding `timeAmount` to `dateNow` on the hour, day, week and months, where it should only be on months slot.